### ISSUE TITLE
quickInfo for variable narrowed to function shows `never`

### ISSUE THREAD

**Issue by CurlerRoo** [microsoft/TypeScript#15844]
**TypeScript Version:**  2.3.2

**Code**

```ts
function foo<T>(t: T) {
    let typeofT = typeof t;
    if (typeofT === "function") {
        t; // "t" is of type "T"
    }
    if (typeof t === "function") {
        t; // "t" is of type "never"
    }
}
```

**Expected behavior:**
"t" should not be narrowed to "never"

**Actual behavior:**
"t" is narrowed to "never"

**Comment by andy-ms** [comment]
First example is because TypeScript won't track that `typeofT` has any special meaning, so it has no effect on type guards.
A simpler way to reproduce this is:
```ts
function foo(t: {}) {
    if (typeof t === "function") {
        t; // Hover shows `t: never`
    }
}
```
You actually won't get errors if you try to call `t`, so this seems more like a quickInfo error.

**Comment by ahejlsberg** [comment]
Seems to me this is a bug. An unconstrained type parameter could be anything, including a function, so it isn't right to narrow to `never`. I think it should be `T & Function`, similar to how we produce `T & string` if you check `typeof t === "string"`.

**Comment by pc-lover** [comment]
@andy-ms 
Calling `t()` is ok. But still, I get an error when calling `t.apply(...)`:
`"Property 'apply' does not exist on type 'never'."`

**Comment by andy-ms** [comment]
@pc-lover Oh dang, #11412 is still an issue. I guess it really is of type `never` and the quickInfo is correct.

**Comment by iisaduan** [comment]
[Playground](https://www.typescriptlang.org/play/?ts=3.3.3#code/GYVwdgxgLglg9mABMOcA8AVAfACigLkQwEpEBvAKEWsQBsBTKRKATwAd65gNEBeZ9p2DMA3FRoxheQVx695iAEShIsBItKUa20YgD0epVEWIYAZ0RcBHJRkXiaBxAD0A-A8QBfD5MTSOVkzy-Mrg0PBgGuQe2lAi+oaKxqYWgYJKYPQAbvQATvY6Tm4UHt6eQA) shows this has been fixed since at least 3.3
